contraint.go

```diff
@@ -134,6 +134,94 @@ func (c CustomStringConstraint) ValidateString(value *string, scope Scope) error
 		CreateViolation()
 }
 
+// ConditionalConstraint is used for conditional validation.
+// Use the When function to initiate a conditional check.
+// If the condition is true, then the constraints passed through the Then function will be applied.
+// Otherwise, the constraints passed through the Else function will apply.
+type ConditionalConstraint struct {
+	condition       bool
+	thenConstraints []Constraint
+	elseConstraints []Constraint
+}
+
+// When function using to initiate a conditional check.
```

Comment:
Грамматика страдает.

When function is used to initiate conditional validation. If the condition is true, then the constraints passed through the Then function will be applied. Otherwise, the constraints passed through the Else function will apply.